# Inference about Population Mean


In [1]:
import pandas as pd
from pandas import DataFrame
import numpy as np
from scipy.stats import t, chi2, norm, f
from math import sqrt


## Inference about Population Mean (μ) When Population Standard Deviation (σ) Is Known
### Weak Law of Large Numbers
For an arbitrarily small positive number $epsilon$, the sample size n can be chosen sufficiently large to make the probability as small as we please so that the average will not differ from the population mean μ by more than $epsilon$. Statement (7.1.8) is sometimes called **the weak law of large numbers**.

<center><img src="Images/Mean1/0.png" width=700></center>

## Margin of Error for Point Estimator of μ
**X_bar** is an unbiased estimators of the population mean μ. Having said this, there still remains
an important question to be answered:<br>
**If the mean μ of a population is unknown, and if we take a random sample from this population and calculate the sample average X_bar, using it as an estimator of μ, how do we know how close our estimator X_bar is to the true value of μ?**<br>
The answer to this question depends on the population size and the sample size.
Let E be the maximum absolute difference between an estimator **X_bar** and the true value of μ, and let 0 <α< 1. Then, if the population is either normal with no restriction on the sample size or a nonnormal infinite population and the sample size is large (n ≥ 30), we say with probability (1 − α) that:

<center><img src="Images/Mean1/6.png" width=300></center>

assuming σ to be known. The quantity E is called the margin of error or bound on the error of estimation. Note that in practice, it is quite common to take α = 0.01, 0.05, or 0.1 since this gives a reasonably high probability (1 − α) that the maximum error of estimation is equal to E.


The result in the equation above is still valid if the population is finite, and the sampling is done with replacement or if the sampling is done without replacement, but the sample size is less than 5% of the population size (n < 0.05N). If the population is finite and the sample size relative to the size of the population is greater than 5%, then we use an extra
factor referred to as the finite population correction factor $\sqrt{\frac{N − n}{N − 1}}$, and the
maximum absolute difference between the estimator X¯ and the true value of μ is
<center><img src="Images/Mean1/7.png" width=350></center>

### Example 1 (Margin of error) 
A manufacturing engineer wants to use the mean of a random sample of size n = 64 to estimate the mean length of the rods being manufactured. If it is known that σ = 0.5 cm, then find the margin of error with 95% probability.

In [8]:
# E = Z_alpha/2 * sigma / sqrt(n)
n = 64
sigma = 0.5
alpha = 0.05
z_alpha = norm.ppf(1-alpha/2)
E = z_alpha * sigma / sqrt(n)
E

0.12249774903375338

## The Central Limit Theorem
<center><img src= "Images/Mean1/1.png" width=750></center>
<center><img src= "Images/Mean1/2.png" width=750></center>
<center><img src= "Images/Mean1/3.png" width=750></center>


### Example 2 (Applying the central limit theorem)
The mean weight of a food entr´ee is μ = 190 g with a standard deviation of 14 g. If a random sample of 49 entr´ees is selected, then find
- The probability that the sample average weight will fall between 186 and 194g.
- The probability that the sample average will be greater than 192 g.

In [2]:
n = 49
mu = 190
sigma = 14

std = sigma / sqrt(n)
std

2.0

So, $\bar{x}$ is normally distributed with mean = 190 and standard deviation = 2.<br>

p(186 < $\bar{x}$ < 194) = p($\frac{186 - 190}{2}$ < z < $\frac{194 - 190}{2}$) = P (-2 < z < 2)

In [34]:
p = norm.cdf(2) - norm.cdf(-2)
print(f"P (186 < xbar < 192) = {p}")

P (186 < xbar < 192) = 0.9544997361036416


In [36]:
# P(X_bar ≥ 192) = P((X_bar − 190)/2 ≥ (192 − 190)/2) = P(Z ≥ 1)
p = 1 - norm.cdf(1)
print(f"P (xbar ≥ 192) = {p}")

P (xbar ≥ 192) = 0.15865525393145707


In [37]:
norm.ppf(0.99)

2.3263478740408408

In [38]:
norm.cdf(2.32634787)

0.9899999998923029

In [49]:
norm.pdf(-10) , norm.pdf(10)

(7.69459862670642e-23, 7.69459862670642e-23)

In [39]:
norm.cdf(10)

1.0

In [53]:
norm.ppf(1)

inf

### Example 3 (Example 1 continued) 
Repeat Example 1 when the sample size is increased from 49 to 64 entr´ees.

In [32]:
n = 64
mu = 190
sigma = 14

std = sigma / sqrt(n)
std

1.75

In [40]:
l = (186 - mu) / std
r = (194 - mu) / std
(l,r)

(-2.2857142857142856, 2.2857142857142856)

In [42]:
p = norm.cdf(r) - norm.cdf(l)
print(f"P (186 < xbar < 192) = {p}")

P (186 < xbar < 192) = 0.9777290210407672


In [44]:
l = (192 - mu) / std
l

1.1428571428571428

In [45]:
# P(X_bar ≥ 192) = P((X_bar − 190)/std ≥ (192 − 190)/std) = P(Z ≥ 1.1428)
p = 1 - norm.cdf(l)
print(f"P (xbar ≥ 192) = {p}")

P (xbar ≥ 192) = 0.12654895447355774


### Example 4 (Applying the central limit theorem to approximate probabilities for a sample mean)
A random sample of 36 reinforcement rods is taken from a production plant that produces these rods with a mean length of μ = 80 cm and a standard deviation of 0.6 cm. Find the approximate probability that the sample mean of the 36 rods falls between 79.85 and 80.15 cm.<br>
**Solution** Since the sample size is large, by the central limit theorem, we know that $\bar{x}$ is approximately normally distributed with mean 80 cm and standard deviation $σ_{\bar{x}} = \frac{σ}{\sqrt{n}} = \frac{0.6}{\sqrt{36}} = 0.1 cm.$

In [6]:
# P(79.85 ≤ X_bar ≤ 80.15) = P((79.85 − 80)/0.1 < (X_bar - 80)/0.1 < (80.15 − 80)/0.1)
# P(−0.15/0.1 ≤ Z ≤ 0.15/0.1) = P(−1.5 ≤ Z ≤ 1.5) = P(Z ≤ 1.5) − P(Z ≤ −1.5) = 0.8664
l = -1.5
r = 1.5
p = norm.cdf(r) - norm.cdf(l)
p

0.8663855974622838

## When the Population is Finite
If the population is finite, the standard error is $\frac{σ}{\sqrt{n}} \sqrt{\frac{N-n}{N-1}}$ where N is the population size and $\sqrt{\frac{N-n}{N-1}}$ is called the **finite population correction factor**. An analysis reveals that if the population size is large relative to the sample size, then the finite population correction factor is close to 1 and can be ignored. <br>

**The correction factor is ignored if either the sampling is done with replacement or  or the population size is at least 20 times larger than the sample size.**

In practice, most applications involve populations that qualify as large because if the population is small, it may be possible to investigate each member of the population, and in so doing, calculate the parameters precisely. As a consequence, the finite population correction factor is usually omitted.


## Interval Estimator of Population Mean
<center><img src= "Images/Mean1/8.png" width=400></center>
<center><img src= "Images/Mean1/9.png" width=700></center>

Note that the confidence interval in Equation (8.3.3) can also be written as X_bar ± E, where E is the margin of error defined in earlier.

## Inference about Population Mean (μ) When Population Standard Deviation (σ) Is Unknown

### The Student t-Distribution
<center><img src= "Images/Mean1/4.png" width=750></center>

### Confidence Interval
<center><img src="Images/Mean1/5.png" width=750></center>
<center><img src="Images/Mean1/10.png" width=750></center>

We remind that the use of Student’s t-distribution to find a confidence interval for the population mean is applicable when the following conditions hold:
1. The population is normal or approximately normal.
2. The sample size is small (n < 30).
3. The population variance is unknown.

For n > 30 we can use the normal distribution instead of t. This case is usually referred to as a large-sample case. Now with the availability of statistical packages, it is very easy to find the t-value for any degrees of freedom, one may continue using the t-distribution instead of switching to z even when n > 30.

## One Tail Confidence Interval
<center><img src="Images/Mean1/11.png" width=750></center>
<center><img src="Images/Mean1/12.png" width=750></center>

### Example 5 (Finding one-sided confidence intervals)
A manufacturing engineer decided to check the efficiency of a new technician hired by the company. She records the time taken by the technician to complete 100 randomly selected jobs and found that in this sample of 100, the average time taken per job was 10 hours with a standard deviation of two hours. Find 95% lower and upper one-sided confidence intervals for μ.

In [13]:
n = 100
x_bar = 10
s = 2
significance = 0.95

# The sample size is large, so I use Z instead of t
z_alpha = norm.ppf(significance)
l = x_bar - z_alpha * s / sqrt(n)
u = x_bar + z_alpha * s / sqrt(n)
l,u

(9.671029274609706, 10.328970725390294)

**Interpretation:** We are 95% confident that the newly hired technician would take a maximum average time of 10.329 hours and a minimum average time of 9.671 hours to complete the job.

### Example 5: Does the mean weekly newspaper collection exceeds 2.0 pounds?
In the near future, nations will likely have to do more to save the environment. Possible
actions include reducing energy use and recycling. Currently, most products manufactured from recycled material are considerably more expensive than those manufactured
from material found in the earth. For example, it is approximately three times as expensive to produce glass bottles from recycled glass than from silica sand, soda ash, and limestone, all plentiful materials mined in numerous countries. It is more expensive to manufacture aluminum cans from recycled cans than from bauxite. Newspapers are an
exception. It can be profitable to recycle newspaper. A major expense is the collection
from homes.
<br>In recent years, many companies have gone into the business of collecting
used newspapers from households and recycling them. A financial analyst for one such
company has recently computed that the firm would make a profit if the mean weekly
newspaper collection from each household exceeded 2.0 pounds.
<br>In a study to determine the feasibility of a recycling plant, a random sample of 148 households was drawn
from a large community, and the weekly weight of newspapers discarded for recycling
for each household was recorded and listed. Do these data provide sufficient evidence to allow the analyst to conclude that a recycling plant would be profitable?

**Solution**
<br>The problem objective is to describe the population of the amounts of newspaper discarded by each household in the population. The data are interval, indicating that the
parameter to be tested is the population mean. Because the financial analyst needs to
determine whether the mean is greater than 2.0 pounds, the alternative hypothesis is:<br>
$ H_1: μ > 2.0 $
<br>As usual, the null hypothesis states that the mean is equal to the value listed in the alternative hypothesis:<br>
$ H_0: μ = 2.0 $

In [7]:
df = pd.read_csv("Data\\recyling newspaper.csv", index_col='i')
df.shape


(148, 1)

The manager believes that the cost of a Type I error (concluding that the mean is greater
than 2 when it isn’t) is quite high. Consequently, he sets the significance level at 99%.

In [23]:
significance = 0.99
mu = 2

n = df.shape[0]
x_bar = np.mean(df)
std = np.std(df, ddof=1).iloc[0]
threshold = t.ppf(significance, df = n-1)
t_stat = (x_bar - mu) / (std/sqrt(n))

print(f"t = {threshold}")
print(f"t statistics = {t_stat}")
if t_stat > threshold:
    print("The null hypothesis is rejected in favor of alternative hypothesis. So the population mean is greater than 2.")
else:
    print("We reject the alternative hypothesis in favor of null hypothesis. Consequently, the population mean in not greater than 2.")

t = 2.3460295783966485
t statistics = 29.338708261513204
The null hypothesis is rejected in favor of alternative hypothesis. So the population mean is greater than 2.


In [40]:
p_value = 1 - t.cdf(threshold, df=n-1)
p_value

0.010000000000340958

The p_value < 0.025 so the null hypothesis is rejceted.

### Example 6: Tax Collected from Audited Returns
In 2014 (the latest year reported), 146,861,217 tax returns were filed in the United States. The Internal Revenue Service (IRS) examined 1,228,117, of them to determine if they were correctly done.<br>
To determine how well the auditors are performing, a random sample of these returns was drawn and the additional tax was reported, which is listed in the tax.csv file. Estimate with 95% confidence the mean additional income tax collected from the 1,228,117 files audited.<br>
**Solution:** The problem objective is to describe the population of additional income tax. The data are interval and hence, the parameter is the population mean μ. The question asks us to estimate this parameter.

In [15]:
df = pd.read_csv("Data\\tax.csv", index_col='i')
df.shape

(192, 1)

In [41]:
significance = 0.95
sl= (1 - significance) / 2
sr = 1 - sl

n = df.shape[0]
x_bar = np.mean(df)
std = np.std(df, ddof=1, axis=0).iloc[0]
tl, tr = t.ppf([sl,sr], df = n-1)
LCL = x_bar + tl * (std/sqrt(n))
UCL = x_bar + tr * (std/sqrt(n))

LCL, UCL

(8886.931964399317, 10167.720743934015)

We estimate that the mean additional tax collected lies between 8, 887 and 10, 168 dolars.
<br>The 95% confidence interval estimate of the total is:

In [19]:
population = 1228117
PLCL, PUCL = (population * LCL, population * UCL)
PLCL, PUCL

(10914192223.322195, 12487150696.87801)

#### Checking the Required Conditions
Histogram of newspaper and tax data to prove that they are normal.